lib/src/definitions.ts

```diff
@@ -59,26 +59,34 @@ export interface Header {
   type: Type;
   optional: boolean;
   examples?: Example[];
+  schemaProps?: SchemaProp[];
 }
 
 export interface PathParam {
   name: string;
   description?: string;
   type: Type;
   examples?: Example[];
+  schemaProps?: SchemaProp[];
 }
 
 export interface Example {
   name: string;
   value: any; // TODO: encapsulate type information
 }
 
+export interface SchemaProp {
+  name: string;
+  value: any; // TODO: encapsulate type information
+}
+
 export interface QueryParam {
   name: string;
   description?: string;
   type: Type;
   optional: boolean;
   examples?: Example[];
+  schemaProps?: SchemaProp[];
```

Comment:
These attributes conceptually are a part of a `Type`. Can we encapsulate these to the `Type` definitions?

Reply:
Yes, if we move it to `Type`, it will be easier